_posts/2020-08-20-relay-pagingation-containers-demystified.markdown

```diff
@@ -0,0 +1,243 @@
+this has left a knowledge gap where Artsy engineers are comfortable _using_ Relay, but they don't totally
+understand how it works.
+
+This is a problem as old as software engineering itself, and it has a simple solution: learn and then teach others.
+We'll be driving a peer learning group centering around Relay, but today we are going to dive into the part of
+Relay that comes up the most in requests for pairing: getting Relay pagination to work. (Note: we're going to use
+plain old Relay and not [relay-hooks](https://github.com/relay-tools/relay-hooks).)
+
+My goal with this post is to show my thought process when trying to learn about, and clean up how we use, Relay
+pagination containers. We'll briefly go over some Relay fundamentals before diving in to a case study on how
+problematic code can get copy-and-pasted throughout your team's code.
+
+Let's back up and talk a little bit about what Relay is and how it works. Relay is a framework that glues React
+components and GraphQL requests together. React components define the data they need from a GraphQL schema in order
+to render themselves, and Relay handles actually fetching GraphQL requests and marshalling data into the React
+component tree. It is very efficient because of build-time optimizations by the Relay compiler.
+
+The simplest use of Relay is a [fragment container](https://relay.dev/docs/en/fragment-container), which is created
+from a React component and a [GraphQL fragment](https://blog.logrocket.com/graphql-fragments-explained/). (We're
+going to skip over how the GraphQL query is made, but
+[here are the docs on query renderers](https://relay.dev/docs/en/query-renderer) if you're curious.)
+
+```js
+class Artist extends React.Component {
+  render() {
+    return <Text>The artist name is ${this.props.artist.name}.</Text>
+  }
+}
+
+export ArtistFragmentContainer = createFragmentContainer(Artist, {
+  artist: graphql` # artist will be passed in as props
+    fragment Artist_artist on Artist { # Relay has strong naming conventions
+      name # Get all the data we want here
+    }
+  `
+})
+```
+
+(At Artsy, we use [TypeScript with Relay](https://github.com/relay-tools/relay-compiler-language-typescript), but
+for this blog post we'll stick to JavaScript.)
+
+So we have a plain React component that gets some props, and a Relay fragment container that wraps it, defining the
+data that the component needs.
+
+There are other types of Relay containers beyond simple fragment containers.
+[Refetch containers](https://relay.dev/docs/en/refetch-container) are like fragment containers except you can
+refetch their contents from your GraphQL server (in response to, for example, user interaction). Using a refetch
+container is very similar to using a plain fragment container. But today, we want to talk about
+[pagination containers](https://relay.dev/docs/en/pagination-container), which use a GraphQL construct called
+_connections_ to show page after page of data.
```

Comment:
<lable> suggestion (non-blocking): Link to pagination blog post

This might be a good place to link to @mzikherman's [post](https://artsy.github.io/blog/2020/01/21/graphql-relay-windowed-pagination/) on GraphQL pagination